README.md

```diff
@@ -53,44 +56,68 @@ run start:sandbox`, but should be re-run whenever DB schema changes are made (ei
+In order to make authorized calls to Figma REST APIs, this application stores Figma user 3LO credentials in the `FigmaOAuth2UserCredentials` table.
 
-1. Start the app.
+### Registering a Figma OAuth application
+
+Follow the steps [here](https://www.figma.com/developers/api#register-oauth2) to register a Figma OAuth app and callback url, and note down the Client ID and Secret.
+
+- Use `APP_URL` for the website URL
+- Use `${APP_URL}/auth/callback` for the callback URL
+
+### Testing the OAuth flow
+
+To test the OAuth 3LO flow and store a Figma users' credentials in the app, you first need to register a Figma application and fill out the `FIGMA_OAUTH_CLIENT_ID` and `FIGMA_OAUTH_CLIENT_SECRET` variables in `.env`.
+
+1. Start the app
 2. Replace the placeholders with actual value and visit the following URL to initiate the OAuth flow.
 
 ```
 https://www.figma.com/oauth?
   client_id=${CLIENT_ID}&
   redirect_uri=${APP_URL}/auth/callback&
-  scope=files:read&
-  state=${ATTLASSIAN_USER_ID}&
+  scope=files:read,file_dev_resources:read,file_dev_resources:write&
+  state=${ATLASSIAN_USER_ID}&
   response_type=code
 ```
 
-You should see a created record in `FigmaUserCredential` table.
+You should see a created record in `FigmaUserCredential` table and hitting `/auth/check3LO?userId=${USER_ID}` should return `{ authorized: true }`
 
 ## Installing the App
 
-To install the app, first ensure both the ngrok tunnel and the app are running, and you've filled out the required
-values in your `.env` file. Then run `npm run jira:installApp`. The app will be installed to the Jira instance specified by
-the `ATLASSIAN_URL` environment variable.
+Before installing, first ensure both the ngrok tunnel and the app are running, and you've filled out the required
+values in your `.env` file.
+
+### Installation script
+
+Run `npm run jira:installApp`. The app will be installed to the Jira instance specified by the `ATLASSIAN_URL` environment variable.
+
+### Manual installs
 
-If you want to install the app in multiple Jira instances, please do it manually. Go to your Jira instances and do
-the following steps:
+If you want to install the app on multiple Jira instances, you can do this manually by performing the following steps:
 
-- From the header menu, select Apps -> Manage your apps.
-- Verify the filter is set to User-installed, and select Settings beneath the User-installed apps table.
-- On the Settings pop-up, add Enable development mode and click Apply. Refresh the page.
-- On the right side of the header, there should now appear a button Upload app. Click it and enter the app URL
-  `/atlassian-connect.json`(`https://${APP_URL}/atlassian-connect.json`)
-- Click Upload.
-- That's it! You're done. 🎉
+1. Visit the **Manage apps** page on your Jira instance (you'll need to be a Jira admin) by visiting this link `https://${JIRA_INSTANCE_BASE_URL}/plugins/servlet/upm` or from the header menu, select **Apps** -> **Manage your apps**.
```

Comment:
`https://${JIRA_INSTANCE_BASE_URL}/plugins/servlet/upm` _might_ be a little bit confusing since we don't have any env vars named `JIRA_INSTANCE_BASE_URL`. You could use `ATLASSIAN_URL` instead, rename `ATLASSIAN_URL` -> `JIRA_INSTANCE_BASE_URL`, or just write this as `https://<your_jira_instance>.atlassian.com/plugins/servlet/upm`